salt/loader.py

```diff
@@ -152,23 +134,93 @@ def _module_dirs(
             ext_type_dirs = "{}_dirs".format(tag)
         if ext_type_dirs in opts:
             ext_type_types.extend(opts[ext_type_dirs])
-        if HAS_PKG_RESOURCES and ext_type_dirs:
-            for entry_point in pkg_resources.iter_entry_points(
-                "salt.loader", ext_type_dirs
-            ):
-                try:
+        if ext_type_dirs:
+            for entry_point in entrypoints.iter_entry_points("salt.loader"):
+                with catch_entry_points_exception(entry_point) as ctx:
                     loaded_entry_point = entry_point.load()
-                    for path in loaded_entry_point():
+                if ctx.exception_caught:
+                    continue
+
+                # Old way of defining loader entry points
```

Comment:
**suggestion**: mention that this is in setup.cfg. When using `setup.py` you'll want:

```
entry_points = '''
    [salt.loader]
    module_dirs = ...
 ```

Reply:
I think this should be addressed in the upcoming salt extensions documentation, not this particular comment.
If one does not know what this is, one should not mess with it and read docs :)